Sources/StreamChat/Query/Filter+ChatChannel.swift

```diff
@@ -4,12 +4,190 @@
 
 import Foundation
 
-extension ChatChannel {
-    func meets(_ filter: Filter<ChannelListFilterScope>) throws -> Bool {
-        // This is a work in progress.
-        // The idea is that in a further iteration we provide a runtime evaluation of the filter to determine
-        // if the channel matches the filter. This is a costly operation, and it is recommended to avoid it.
-        // The recommended approach is to pass a `filter` block when initializing a `ChatChannelListController`
-        true
+extension Filter where Scope == ChannelListFilterScope {
+    /// If it can be translated, this will return
+    /// an NSPredicate instance that is equivalent
+    /// to the current filter.
+    ///
+    /// For now it's limited to ChannelList as it's not
+    /// needed anywhere else
+    ///
+    /// The predicate will be automatically be used
+    /// by the ChannelDTO to create the
+    /// fetchRequest.
+    ///
+    /// - Important:
+    /// The behaviour of the ChannelDTO, to include or not
+    /// the predicate in the fetchRequest, it's controlled by
+    /// `StreamRuntimeCheck.isChannelLocalFilteringEnabled`
+    var predicate: NSPredicate? {
+        guard let `operator` = FilterOperator(rawValue: `operator`) else {
```

Comment:
Nit: In this case we could have used just "op" to avoid always typing `` whenever we used this variable